crates/viewer/re_space_view_map/src/map_space_view.rs

```diff
@@ -0,0 +1,334 @@
+                map_state.map_memory.follow_my_position();
+            }
+        });
+
+        Ok(())
+    }
+
+    fn ui(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        state: &mut dyn SpaceViewState,
+
+        query: &ViewQuery<'_>,
+        system_output: SystemExecutionOutput,
+    ) -> Result<(), SpaceViewSystemExecutionError> {
+        let state = state.downcast_mut::<MapSpaceViewState>()?;
+        let map_viz_system = system_output.view_systems.get::<MapVisualizerSystem>()?;
+
+        let blueprint_db = ctx.blueprint_db();
+        let view_id = query.space_view_id;
+        let map_options =
+            ViewProperty::from_archetype::<MapOptions>(blueprint_db, ctx.blueprint_query, view_id);
+        let map_provider = map_options.component_or_fallback::<MapProvider>(ctx, self, state)?;
+        let zoom_level = map_options
+            .component_or_fallback::<ZoomLevel>(ctx, self, state)?
+            .0;
+        let access_token = map_options
+            .component_or_fallback::<Secret>(ctx, self, state)?
+            .0
+            .to_string();
+
+        if state.map_memory.set_zoom(*zoom_level).is_err() {
+            re_log::warn_once!(
+                "Failed to set zoom level for map. Zoom level should be between zero and 22"
+            );
+        };
+
+        // if state changed let's update it from the blueprint
+        if state.selected_provider != map_provider || access_token != state.access_token {
+            state.tiles = None;
+            state.selected_provider = map_provider;
+            state.access_token = access_token;
+        }
+
+        let (tiles, map_memory) = match state.ensure_and_get_mut_refs(ui.ctx()) {
+            Ok(refs) => refs,
+            Err(err) => return Err(err),
+        };
+
+        egui::Frame::default().show(ui, |ui| {
+            let some_tiles_manager: Option<&mut dyn Tiles> = Some(tiles);
+            let map_widget = ui.add(
+                Map::new(
+                    some_tiles_manager,
+                    map_memory,
+                    map_viz_system
+                        .map_entries
+                        .first()
+                        .unwrap_or(&MapEntry::default())
+                        .position,
+                )
+                .with_plugin(PositionsOnMap {
+                    positions: map_viz_system.map_entries.clone(),
+                }),
+            );
+
+            map_widget.double_clicked().then(|| {
+                map_memory.follow_my_position();
+            });
+
+            let map_pos = map_widget.rect;
+            let window_id = query.space_view_id.uuid().to_string();
+            map_windows::zoom(ui, &window_id, &map_pos, map_memory);
+            map_windows::acknowledge(ui, &window_id, &map_pos, tiles.attribution());
+
+            // update blueprint if zoom level changed from ui
+            if map_memory.zoom() != *zoom_level {
```

Comment:
see my other note, would be great to store the map position in the blueprint as well